tools/cdb2sockpool/cdb2sockpool.c

```diff
@@ -220,11 +220,11 @@ int recvall(int fd, void *bufp, int len)
 
     while (bytesleft) {
         rc = recv(fd, buf + off, bytesleft, MSG_WAITALL);
-        if (rc == -1) {
-            if (rc == EINTR || rc == EAGAIN)
-                continue;
+        if (rc == EINTR || rc == EAGAIN)
```

Comment:
I think this should be `if (rc ==-1 && errno == EINTR || errno == EAGAIN)`

Reply:
yes, thanks. Fixed.